src/components/timeline/LayerDiscrete.svelte

```diff
@@ -257,41 +290,48 @@
     }
   }
 
-  function getDirectivesAndSpansForOffset(offsetX: number, offsetY: number) {
+  function getItemsForOffset(offsetX: number, offsetY: number) {
+    const externalEvents = searchQuadtreeRect<ExternalEvent>(
+      quadtreeExternalEvents,
+      offsetX,
+      offsetY,
+      discreteOptions.height,
+      maxExternalEventWidth,
+      visibleExternalEventsById,
+    );
     const activityDirectives = searchQuadtreeRect<ActivityDirective>(
       quadtreeActivityDirectives,
       offsetX,
       offsetY,
-      activityOptions.activityHeight,
+      discreteOptions.height,
       maxActivityWidth,
       visibleActivityDirectivesById,
     );
     const spans = searchQuadtreeRect<Span>(
       quadtreeSpans,
       offsetX,
       offsetY,
-      activityOptions.activityHeight,
+      discreteOptions.height,
       maxActivityWidth,
       visibleSpansById,
     );
-    return { activityDirectives, spans };
+    return { activityDirectives, externalEvents, spans };
   }
 
   function onMousedown(e: MouseEvent | undefined): void {
     // Do not process events if meta/ctrl is pressed to avoid interaction conflicts with zoom/pan
     if (e && timelineInteractionMode === TimelineInteractionMode.Interact && e.button !== 1) {
       const { offsetX, offsetY } = e;
-      const { activityDirectives, spans } = getDirectivesAndSpansForOffset(offsetX, offsetY);
+      const { activityDirectives, spans, externalEvents } = getItemsForOffset(offsetX, offsetY);
 
       /**
        * The setTimeout is needed to prevent a race condition with mousedown events and change events.
-       * Without the setTimeout, mousedown events happen before change events in the activity directive form.
```

Comment:
Can we keep these old comments especially since right now there are no change events being fired by the external event form?